src/Storage/Repository/IBlobRepository.cs

```diff
@@ -16,28 +16,34 @@ public interface IBlobRepository
         /// <param name="org">The application owner id.</param>
         /// <param name="stream">Data to be written to blob storage.</param>
         /// <param name="blobStoragePath">Path to save the stream to in blob storage.</param>
+        /// <param name="storageContainerNumber">Alternate number to append to container name</param>
         /// <returns>The size of the blob.</returns>
-        Task<(long ContentLength, DateTimeOffset LastModified)> WriteBlob(string org, Stream stream, string blobStoragePath);
+        Task<(long ContentLength, DateTimeOffset LastModified)> WriteBlob(string org, Stream stream, string blobStoragePath, int? storageContainerNumber);
 
         /// <summary>
         /// Reads a data file from blob storage
         /// </summary>
         /// <param name="org">The application owner id.</param>
         /// <param name="blobStoragePath">Path to be file to read blob storage.</param>
+        /// <param name="storageContainerNumber">Alternate number to append to container name</param>
         /// <returns>The stream with the file</returns>
-        Task<Stream> ReadBlob(string org, string blobStoragePath);
+        Task<Stream> ReadBlob(string org, string blobStoragePath, int? storageContainerNumber);
 
         /// <summary>
         /// Deletes the blob element permanently
         /// </summary>
         /// <param name="org">The application owner id.</param>
         /// <param name="blobStoragePath">Path to the file to delete.</param>
+        /// <param name="storageContainerNumber">Alternate number to append to container name</param>
         /// <returns>A value indicating whether the delete was successful.</returns>
-        Task<bool> DeleteBlob(string org, string blobStoragePath);
+        Task<bool> DeleteBlob(string org, string blobStoragePath, int? storageContainerNumber);
 
         /// <summary>
         /// Deletes the blob elements for an instance permanently
         /// </summary>
-        Task<bool> DeleteDataBlobs(Instance instance);
+        /// <param name="instance">The instance to delete from</param>
+        /// <param name="storageContainerNumber">Alternate number to append to container name</param>
+        /// <returns>A value indicating whether the delete was successful.</returns>/// 
+        Task<bool> DeleteDataBlobs(Instance instance, int? storageContainerNumber);
```

Comment:
I prefer more specific parameters. DeleteBlobs doesn't need the full instance. Limit it to org, app and id. Make sure the id is without partyid.

Reply:
???

Reply:
```suggestion
        Task<bool> DeleteDataBlobs(org, app, id, int? storageContainerNumber);
```

Reply:
Ref. standup - we don't introduce the suggested change in this PR because it's not related to the PR